src/measure_theory/covering/liminf_limsup.lean

```diff
@@ -0,0 +1,196 @@
+/-
+Copyright (c) 2022 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import measure_theory.covering.density_theorem
+
+/-!
+# Liminf, limsup, and doubling measures.
+
+This file is a place to collect lemmas about liminf and limsup for subsets of a metric space
+carrying a doubling measure.
+
+## Main results:
+
+ * `blimsup_cthickening_mul_ae_eq`: the limsup of the closed thickening of a sequence of compact
+   subsets is unchanged if the sequence of distances is multiplied by a positive scale factor.
+
+-/
+
+open set filter metric measure_theory
+open_locale nnreal ennreal topological_space
+
+variables {α : Type*} [metric_space α] [proper_space α] [measurable_space α] [borel_space α]
```

Comment:
I'm still not sure about the `proper_space` assumption. Where is it important in the proof? (Note that, in general, `cthickening r s` contains all the balls `closed_ball x r` for `x` in `s`, but it is also contained in the union of `closed_ball x r'` over `x` in `s`, for any `r' > r`. And since you are allowed to wiggle a little bit your radii, I guess everything should be fine).

Reply:
There was one other superficial use of `proper_space` which I have removed.

I will return to tackle this one later today and then I should be able to assume `sigma_compact_space` instead of `proper_space`.

Reply:
Are you sure you need anything? My guess is you don't even need `sigma_compact_space`.

Reply:
I need `sigma_compact_space` because I need to use the density theorem `is_doubling_measure.ae_tendsto_measure_inter_div`.

Reply:
Ah, that's true there are some assumptions for Lebesgue density theorem. Then `sigma_compact_space` is perfect.

Reply:
I have now pushed a commit which relaxes `proper_space` to `sigma_compact_space`.